doc/syntax.rst

```diff
@@ -289,6 +298,39 @@ as an array:
 
 * each row represents a shot/run
 
+For-loops
+~~~~~~~~~
+
+Similar to Python, for-loops can be declared using the ``for ... in ...`` syntax, followed by lines
+of indented statements. Notice that there is no colon (``:``) at the end of the for-statement. The for-loop
+variable type must be declared followed by either a list of values, of the specified type, or a
+range using the syntax ``from:to:step``.
+
+For example:
+
+.. code-block:: python
+
+    for int i in [0, 2, 1, 0, 2, 1]
+        MZgate(phases[i], phases[i+1]) | [i, i+1]
+
+where ``phases`` could be an array declared above, or:
+
+.. code-block:: python
+
+    for int m in 2:10:2
+        MeasureX | m
+
+measuring over modes 2, 4, 6 and 8.
+
+.. note::
+
+  Currently, the following are not supported:
+
+  * Nested for-loops; only single for-loops are allowed
+
+  * Looping through arrays, e.g. ``for int i in phases``
```

Comment:
```suggestion
  * Nested for-loops; only single for-loops are allowed,

  * Looping through arrays, e.g. ``for int i in phases``.
```

Reply:
(I imagine the second bullet point would be quite easy to add support for at some point? 🙂 )

Reply:
Yes, that should be doable; I just haven't had a chance to look into it yet. :slightly_smiling_face: I'll remember to update the docs when this is supported.